setup.py

```diff
@@ -101,7 +101,16 @@ def is_platform_mac():
     'include_package_data': True,
     'setup_requires': ['pytest-runner', 'numpy>=1.15'],
     'install_requires': ['numpy>=1.15', 'six>=1.10', 'scipy', 'astropy>=2.0', 'h5py'],
-    'tests_require': ['pytest'],
+    'tests_require': ['pytest', 'pytest-cases'],
+    'extras_require': {
+        'casa': ['python-casacore'],
+        'healpix': ['astropy-healpix'],
+        'cst': ['pyyaml'],
+        'all': ['python-casacore', 'pyyaml', 'astropy-healpix'],
+        'dev': ['python-casacore', 'pyyaml', 'astropy-healpix', 'pytest',
+                'pytest-cases', 'pytest-cov', 'coverage', 'flake8',
+                'flake8-pytest', 'sphinx', 'pypandoc']
```

Comment:
The difference between `dev` and `all` is both test and docs related, do we want a `test` and a `doc` option?

Reply:
I think we could possibly refactor this a bit, so that we could easily just add lists here. Something like:

```
casa_rq = ['python-casacore']
healpix_rq = ['astropy-healpix']
cst_rq = ['pyyaml']
all = casa_rq + healpix_rq + cst_rq
test = ['pytest-cases', 'pytest-cov', 'coverage', 'flake8', 'flake8-pytest']
doc = ['sphinx', 'pypandoc']
dev = all + test + doc
```

Reply:
sure, I can do that. I had thought about lists that we add but worried a little about making it less readable. But that may not be important.